packages/amplify_auth_cognito/example/lib/main.dart

```diff
@@ -137,6 +115,20 @@ class _MyAppState extends State<MyApp> {
     return session.isSignedIn;
   }
 
+  Future<void> _deleteUser() async {
+    try {
+      await Amplify.Auth.deleteUser();
+      changeDisplay('SHOW_SIGN_IN');
+    } on AmplifyException catch (e) {
+      setState(() {
+        _error = e;
+      });
+      print(e);
+    } on UnimplementedError catch (e) {
+      print(e.message);
+    }
```

Comment:
Catching errors is an anti-pattern, especially here. The correct way to handle this is from a UX perspective is to not even show the delete user button on Android. And then have a check at the start of this function making it a no-op on Android. The goal is that the `UnimplementedError` is never thrown in the first place.

Reply:
I had this here so that the Android behavior could be tested in the example app; but since we have integ tests for this, I'm fine with removing it.

Reply:
Okay. My point is that we need to model how best to use this API here.